plugins/extraction/ocaml.ml

```diff
@@ -697,11 +697,35 @@ let is_short = function MEident _ | MEapply _ -> true | _ -> false
 
 let rec pp_structure_elem = function
   | (l,SEdecl d) ->
-     (match Common.get_duplicate (top_visible_mp ()) l with
-      | None -> pp_decl d
-      | Some ren ->
-         hov 1 (str ("module "^ren^" = struct") ++ fnl () ++ pp_decl d) ++
-         fnl () ++ str "end" ++ fnl () ++ str ("include "^ren))
+    let duplicates =
+      match Common.get_duplicate (top_visible_mp ()) l with
+      | None -> []
+      | Some ren -> [ren] in
+    (* Records may have shadowed fields, and they also need to be double-checked for duplicates. *)
+    let duplicates =
+      match d with
+      | Dind (_, i) ->
+        (match i.ind_kind with
+         | Record fields ->
```

Comment:
Note that I only check for the record case, but this issue suggests that it would be possible to be shadowed in an inductive definition too: it might be worth adding a special case here.